### ISSUE TITLE
Babel 2.9 was failing to build docs

### ISSUE THREAD

**Issue by hgrecco** [hgrecco/pint#1219]
In 3484cdec44ae15f4f3675372c384d9d6caaa3cf9 Babel was pinned in travis to 2.8 as building the docs was failing with 2.9

```
**********************************************************************
File "tutorial.rst", line 418, in default
Failed example:
    "%s" % accel
Exception raised:
    Traceback (most recent call last):
      File "/Users/grecco/anaconda3/envs/pint-doctest/lib/python3.7/doctest.py", line 1337, in __run
        compileflags, 1), test.globs)
      File "<doctest default[2]>", line 1, in <module>
        "%s" % accel
      File "/Users/grecco/Documents/code/pint/pint/quantity.py", line 286, in __str__
        return self.format_babel()
      File "/Users/grecco/Documents/code/pint/pint/quantity.py", line 431, in format_babel
        obj.units.format_babel(spec, **kwspec),
      File "/Users/grecco/Documents/code/pint/pint/unit.py", line 112, in format_babel
        return "%s" % (units.format_babel(spec, **kwspec))
      File "/Users/grecco/Documents/code/pint/pint/util.py", line 452, in format_babel
        return format_unit(self, spec, **kwspec)
      File "/Users/grecco/Documents/code/pint/pint/formatting.py", line 284, in format_unit
        return formatter(unit.items(), **fmt)
      File "/Users/grecco/Documents/code/pint/pint/formatting.py", line 237, in formatter
        neg_ret = _join(division_fmt, neg_terms)
      File "/Users/grecco/Documents/code/pint/pint/formatting.py", line 41, in _join
        if not __JOIN_REG_EXP.search(fmt):
    TypeError: expected string or bytes-like object
**********************************************************************
```

For some reason `fmt` is a `babel.localedata.LocaleDataDict`object.

**Comment by jules-ch** [comment]
Investigating a bit, we rely on private part of the babel API.

the culprit is here : https://github.com/python-babel/babel/commit/462444f35756551f83809dab7f7424a907033a9e

In the formatter, we need to change this line
https://github.com/hgrecco/pint/blob/cdf0a50f49c6ad55a77b312166d3803f17d3e618/pint/formatting.py#L204-L206
 to :

```python

            division_fmt = (
                compound_unit_patterns.get("per", {})
                .get(babel_length, {})
                .get("compound", division_fmt)
            )
```

**Comment by jules-ch** [comment]
babel 2.9 is still not supported, CI has been fixed though and maximal version for babel's been specified in setup.cfg

**Comment by mbanck** [comment]
I think just pinning locally to babel-.2.8 won't cut it, Debian e.g. has (maybe somewhat hastily) moved on to babel-2.10 and now pint (and with it a lot of reverse dependencies) is about to be removed from the upcoming release.

The change in #1401 seems to fix this (eb4e13428a in particular), but that PR was abandoned. Is that something Debian can pick up for now, or was it deemed an insufficient fix?
